e2e/suite_test.go

```diff
@@ -142,13 +147,24 @@ func runScenario(ctx context.Context, t *testing.T, r *mrand.Rand, opts *scenari
 	// Only perform node readiness/pod-related checks when VMSS creation succeeded
 	if vmssSucceeded {
 		t.Log("vmss creation succeded, proceeding with node readiness and pod checks...")
-		if err = validateNodeHealth(ctx, t, opts.kube, vmssName); err != nil {
+		nodeName, err := validateNodeHealth(ctx, t, opts.kube, vmssName)
+		if err != nil {
 			t.Fatal(err)
 		}
 
+		if opts.nbc.AgentPoolProfile.WorkloadRuntime == datamodel.WasmWasi {
+			t.Log("wasm scenario: running wasm validation...")
+			if err := ensureWasmRuntimeClasses(ctx, opts.kube); err != nil {
```

Comment:
okay for now, but wonder if we should have some per-test setup or something

Reply:
yep I'm not a huge fan of this but figured it would be the simplest way forward for now - we can probably also abstract out k8s-level validation to per-scenario config as you suggested below